temporalcli/commands.taskqueue.go

```diff
@@ -10,31 +10,243 @@ import (
+	case client.BuildIDTaskReachabilityReachable:
+		return "reachable", nil
+	case client.BuildIDTaskReachabilityClosedWorkflowsOnly:
+		return "closedWorkflowsOnly", nil
+	case client.BuildIDTaskReachabilityUnreachable:
+		return "unreachable", nil
+	default:
+		return "", fmt.Errorf("unrecognized reachability type: %d", reachability)
+	}
+}
+
+func descriptionToReachabilityRows(taskQueueDescription client.TaskQueueDescription) ([]reachabilityRowType, error) {
+	var rRows []reachabilityRowType
+	// Unversioned queue first
+	val, ok := taskQueueDescription.VersionsInfo[client.UnversionedBuildID]
+	if ok {
+		reachability, err := reachabilityToStr(val.TaskReachability)
+		if err != nil {
+			return nil, err
+		}
+		rRows = append(rRows, reachabilityRowType{
+			BuildID:      taskQueueUnversioned,
+			Reachability: reachability,
+		})
+	}
+	for k, val := range taskQueueDescription.VersionsInfo {
+		if k != client.UnversionedBuildID {
+			reachability, err := reachabilityToStr(val.TaskReachability)
+			if err != nil {
+				return nil, err
+			}
+			rRows = append(rRows, reachabilityRowType{
+				BuildID:      k,
+				Reachability: reachability,
+			})
+		}
+	}
+	return rRows, nil
+}
+
+func taskQueueTypeToStr(taskQueueType client.TaskQueueType) (string, error) {
+	switch taskQueueType {
+	case client.TaskQueueTypeUnspecified:
+		return "unspecified", nil
+	case client.TaskQueueTypeWorkflow:
+		return "workflow", nil
+	case client.TaskQueueTypeActivity:
+		return "activity", nil
+	case client.TaskQueueTypeNexus:
+		return "nexus", nil
+	default:
+		return "", fmt.Errorf("unrecognized task queue type: %d", taskQueueType)
+	}
+}
+
+func buildIDToPollerRows(pRows []pollerRowType, buildID string, typesInfo map[client.TaskQueueType]client.TaskQueueTypeInfo) ([]pollerRowType, error) {
+	for t, info := range typesInfo {
+		taskQueueType, err := taskQueueTypeToStr(t)
+		if err != nil {
+			return pRows, err
+		}
+		for _, p := range info.Pollers {
+			pRows = append(pRows, pollerRowType{
+				BuildID:        buildID,
+				TaskQueueType:  taskQueueType,
+				Identity:       p.Identity,
+				LastAccessTime: p.LastAccessTime,
+				RatePerSecond:  p.RatePerSecond,
+			})
+		}
+	}
+	return pRows, nil
+}
+
+func descriptionToPollerRows(taskQueueDescription client.TaskQueueDescription) ([]pollerRowType, error) {
+	var pRows []pollerRowType
+	var err error
+	// Unversioned queue first
+	val, ok := taskQueueDescription.VersionsInfo[client.UnversionedBuildID]
+	if ok {
+		pRows, err = buildIDToPollerRows(pRows, taskQueueUnversioned, val.TypesInfo)
+		if err != nil {
+			return nil, err
+		}
+	}
+	for k, val := range taskQueueDescription.VersionsInfo {
+		if k != client.UnversionedBuildID {
+			pRows, err = buildIDToPollerRows(pRows, k, val.TypesInfo)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return pRows, nil
+}
+
+func descriptionToRows(taskQueueDescription client.TaskQueueDescription, reportReachability bool) (taskQueueDescriptionType, error) {
```

Comment:
This name is definitely too general for a large shared package. Also, if a function is called once, consider just inlining it instead of all of these tiny functions (not very common in Go to have a bunch of tiny single-use functions)

Reply:
renamed to `taskQueueDescriptionToRows`